fs_mark/fs_mark_clean.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Cleanup after fs_mark (executed with files kept).
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <dirent.h>
+#include <string.h>
+#include <limits.h>
+
+int remove_dir_recursive(const char *dirPath)
+{
+	DIR *dir;
+	char entryPath[PATH_MAX];
+	struct dirent *entry;
+	int (*remove_func)(const char *);
+	int ret, try = 0;
+
+	dir = opendir(dirPath);
+	if (dir == NULL) {
+		fprintf(stderr, "Opendir() failed errno: %s\n", strerror(errno));
+		return -1;
+	}
+
+	if (dir) {
+		while ((entry = readdir(dir)) != NULL) {
+			if (!strcmp(".", entry->d_name) || !strcmp("..", entry->d_name)) {
+				continue;
+			}
+			sprintf(entryPath, "%s/%s", dirPath, entry->d_name);
+			remove_func = entry->d_type == DT_DIR ? remove_dir_recursive : remove;
```

Comment:
I am not sure about using recursion - we rather avoid it - the stack usage is very high then. However maybe we could leave it in such use case.

Reply:
Stack of this function is maybe 400 bytes long so it would need many levels of nested directories to exceed its size. At this moment there is only 1 level of nested directories.